controller/hydrator_dependencies_test.go

```diff
@@ -0,0 +1,77 @@
+package controller
+
+import (
+	"encoding/json"
+	"errors"
+	"github.com/argoproj/argo-cd/v3/common"
+	"github.com/argoproj/argo-cd/v3/pkg/apis/application/v1alpha1"
+	"github.com/argoproj/argo-cd/v3/reposerver/apiclient"
+	"github.com/argoproj/argo-cd/v3/test"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"testing"
+	"time"
+)
+
+func TestGetRepoObjs(t *testing.T) {
+	cm := test.NewConfigMap()
+	cm.SetAnnotations(map[string]string{
+		"custom-annotation":             "custom-value",
+		common.AnnotationInstallationID: "id",     // tracking annotation should be removed
```

Comment:
The main purpose of setting this is just to keep track of whether these annotations gets removed or not. Although  Argo CD internally sets these tracking annotation and outputs the resource without those annotations but since we want to make sure those annotations are actually getting removed (that's the purpose of this test), we are adding it here to later test for their removal.